.github/workflows/golang.yml

```diff
@@ -33,13 +33,23 @@ jobs:
       if: needs.check-changed-files.outputs.golang == 'true'
       run: if [[ $(go list ./... | xargs go fmt) ]]; then echo "Please run go fmt"; exit 1; fi
 
-  build-linux:
-    name: Build on Linux
+  build-curves-linux:
+    name: Build curves on Linux
     runs-on: [self-hosted, Linux, X64, icicle]
     needs: [check-changed-files, check-format]
     strategy:
       matrix:
-        curve: [bn254, bls12_381, bls12_377, bw6_761]
+        curve: 
+          - name: bn254
+            build_args: -g2 -ecntt
+          - name: bls12_381
+            build_args: -g2 -ecntt
+          - name: bls12_377
+            build_args: -g2 -ecntt
+          - name: bw6_761
+            build_args: -g2 -ecntt
+          - name: grumpkin
+            build_args:
```

Comment:
TODO Roman: implement the same for our c++ workflow (rn it just doesn't test grumpkin at all)